db/db.go

```diff
@@ -169,6 +221,29 @@ func (db *Database) InsertState(path string, versionID string, sf *statefile.Fil
 	return nil
 }
 
+// UpdateState update a Terraform State in the Database with Lineage foreign constraint
+// It will also insert Lineage entry in the db if needed
+func (db *Database) UpdateState(st types.State) error {
```

Comment:
This function is only useful for the migration, right?

Reply:
Atm yes

Reply:
OK. I'm wondering if a comment on this could be good, so it's clear that we don't normally update states in place (since they're supposed to be immutable objects in the DB)

Reply:
This is not a bad idea, I will add this detail to avoid any ambiguity :+1: